homeassistant/components/rainforest_raven/config_flow.py

```diff
@@ -0,0 +1,155 @@
+"""Config flow for Rainforest RAVEn devices."""
+from __future__ import annotations
+
+import asyncio
+from typing import Any
+
+from aioraven.data import MeterType
+from aioraven.device import RAVEnConnectionError
+from aioraven.serial import RAVEnSerialDevice
+import serial.tools.list_ports
+from serial.tools.list_ports_common import ListPortInfo
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import usb
+from homeassistant.const import CONF_DEVICE, CONF_MAC, CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.selector import (
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
+
+from .const import DEFAULT_NAME, DOMAIN
+
+
+def _format_id(value: str | int) -> str:
+    if isinstance(value, str):
+        return value
+    return f"{value or 0:04X}"
+
+
+def _generate_unique_id(info: ListPortInfo | usb.UsbServiceInfo) -> str:
+    """Generate unique id from usb attributes."""
+    return f"{_format_id(info.vid)}:{_format_id(info.pid)}_{info.serial_number}_{info.manufacturer}_{info.description}"
```

Comment:
Please split this, for example:
```suggestion
    return (
        f"{_format_id(info.vid)}:{_format_id(info.pid)}_{info.serial_number}"
        f"_{info.manufacturer}_{info.description}"
    )
```

Reply:
Implemented in c142a16.